### PR TITLE
Update delivery step

### LINKED ISSUE #83: Functions associated with 'Data Release 1.0' are out of date
The data release step has been renamed so we should update these functions.

### PR DESCRIPTION
**Pull request by tcezard**
This PR add the new delivery step name and remove an unused function
closes #83

### CODE REVIEW

File: tests/test_clarity.py [hunk 1]
```diff
@@ -253,11 +253,6 @@ def test_get_samples_sequenced_with(self, mocked_get_sample, mocked_containers,
         mocked_names_from_plate.assert_any_call('this')
         mocked_names_from_plate.assert_any_call('that')
 
-    @patched_lims('get_processes', [FakeProcess])
-    def test_get_released_samples(self, mocked_lims):
-        assert clarity.get_released_samples() == ['that', 'this']
-        mocked_lims.assert_called_with(type='Data Release EG 1.0')
-
     @patched_clarity('get_sample', Mock(artifact=Mock(id='an_artifact_id')))
     @patched_lims('get_processes', side_effect=[[FakeProcess], [FakeProcess, FakeProcess(date_run='another_date')]])
```
Comment (mwhamgenomics) [inline tests/test_clarity.py]: get_sample_release_date now does two calls to get_processes, so the side_effect here gets exhausted early. The following fixes it:

```python
@patched_lims('get_processes', side_effect=[[FakeProcess], [], [], [FakeProcess, FakeProcess(date_run='another_date')]])
def test_get_sample_release_date(self, mocked_get_procs, mocked_get_sample):
    assert clarity.get_sample_release_date('a_sample_name') == 'a_date'
    for step_name in ['Data Release EG 1.0', 'Data Release EG 2.0']:
        mocked_get_procs.assert_any_call(type=step_name, inputartifactlimsid='an_artifact_id')
...
```

**Review by mwhamgenomics — CHANGES_REQUESTED** [review]
Unit test needs fixing. Also, shouldn't this be against Release_v0.12?

**Review by lukbut — CHANGES_REQUESTED** [review]
As @mwhamgenomics said, the unit tests need to be updated as does the target branch.

File: tests/test_clarity.py (hunk 1 above)
Comment (tcezard) [reply tests/test_clarity.py]: Thanks
That should be done now

**Review by lukbut — APPROVED** [review]

**Review by mwhamgenomics — APPROVED** [review]

**Pull request merged**